File: app/src/main/java/com/confessionsearchapptest/release1/data/notes/InsertAsync.java
package com.confessionsearchapptest.release1.data.notes;

import android.os.AsyncTask;

public class InsertAsync extends AsyncTask<Notes, Void, Void> {

    private final NoteDao mNotesDao;

    public InsertAsync(NoteDao dao) {
        mNotesDao = dao;
    }

    @Override
    protected Void doInBackground(Notes... notes) {
        mNotesDao.insertNotes(notes);
        return null;
    }}

File: app/src/main/java/com/confessionsearchapptest/release1/data/notes/InsertAsync.kt
package com.confessionsearchapptest.release1.data.notes

import android.os.AsyncTask
import com.confessionsearchapptest.release1.async.CoroutinesAsyncTask

class InsertAsync(private val mNotesDao: NoteDao) : CoroutinesAsyncTask<Notes?, Void?, Void?>("insert") {
    override fun doInBackground(vararg notes: Notes?): Void? {
        mNotesDao.insertNotes(*notes)
        return null
    }
}